contrib/vcl/source/BUILD

```diff
@@ -28,37 +29,63 @@ cc_library(
     visibility = ["//visibility:public"],
 )
 
-genrule(
+envoy_cmake(
     name = "build",
-    srcs = [
-        "@com_github_fdio_vpp_vcl//:all",
+    build_data = [requirement("ply")],
+    cache_entries = {
+        "CMAKE_BUILD_TYPE": "Release",
+        "VPP_API_TEST_BUILTIN": "OFF",
+        "BUILD_SHARED_LIBS": "OFF",
+        "CMAKE_ENABLE_TESTING": "OFF",
+        "CMAKE_ENABLE_EXPORTS": "OFF",
+    },
+    # build_args = ["--", "-C", ".", "vppcom"],
+    copts = ["-Wno-unused-variable"],
+    default_cache_entries = {},
+    env = {
+        "PLYPATHS": "$(locations %s)" % requirement("ply"),
+    },
+    lib_source = "@com_github_fdio_vpp_vcl//:all",
+    linkopts = ["-Wno-unused-variable"],
+    out_static_libs = [
+        "libvppcom.a",
+        "libvppinfra.a",
+        "libsvm.a",
+        "libvlibmemoryclient.a",
+    ],
+    postfix_script = """
+        mkdir -p $INSTALLDIR/lib/external $INSTALLDIR/include/external \
+        && find . -name "*.a" | xargs -I{} cp -a {} $INSTALLDIR/lib/ \
+        && find . -name "*.h" | xargs -I{} cp -a {} $INSTALLDIR/include
+    """,
+    tags = [
+        "cpu:16",
+        "skip_on_windows",
     ],
+    # tags = ["no-remote", "skip_on_windows"],
+    targets = [
+        "",
+        "install",
+    ],
+    working_directory = "src",
+)
+
+genrule(
+    name = "build_files",
     outs = [
-        "external/libvppcom.a",
-        "external/libvppinfra.a",
         "external/libsvm.a",
         "external/libvlibmemoryclient.a",
+        "external/libvppcom.a",
+        "external/libvppinfra.a",
         "external/vppcom.h",
     ],
     cmd = """
```

Comment:
Could we maybe leverage something like [this](https://github.com/envoyproxy/envoy/pull/18149/commits/7f589cc7eff951d66f626589270d303e71707ded#diff-a6a4eb9af2d1f92545242b7adfe0eda8d96cca4f8d6f2b55902846da59e8502b) first approach to building vppcom?